docs/development/apifirst.md

```diff
@@ -0,0 +1,116 @@
+---
+sidebar_position: 6
+---
+
+# REST APIs
+
+Building [RESTFull](https://www.redhat.com/en/topics/api/what-is-a-rest-api) APIs is a typical use
+case for Node.js. There are two typical approaches:
+
+* API First - define the API, use tools to help scaffold and then fill in the implementation.
+* Code First - implemhdawson-patch-1ment the APIs and then document based on the API exposed.
+
+The team's experience is that the API first approach based on [OpenAPI](https://swagger.io/specification/)
+provides benefits in both initial implementation and maintenance and our recommended packages and
+guidance is based on that approach, particularly if one or more of the following are true:
+
+- New project without existing API
+- Client side and backend are developed by different teams that need way to communicate API changes
+- Development on clients and backends starts around the same time giving developers ability to mock API based on OpenAPI spec
+
+> NOTE: This section of the reference arcahitecture focusses on building RESTfull APIs. 
+For GraphQL please refer to the [GraphQL Guide][] section.
+
+## "API-first" approach
+
+With the API-first approach, designing the API is the first priority before writing any code. Design of the API involves thorough thinking and planning through collaboration with different teams (both client and backend side). This results in high-level documentation describing the intent of the API and ability to build client without waiting for server to be finished.
+
+This API contract acts as a central draft keeping all your team members aligned on what your API’s objectives are and how your API’s resources are exposed. The finalization of the contract allows the team to build the interface of the application.
+
+After this, the cross-functional teams rely on this interface to build the rest of the application independent of each other. In practice, teams can generate backends stubs and fully functional client libraries to avoid deviation from specification set in the OpenAPI spec file.
+
+## Code First vs API first
+
+Code first approach provides libraries that understand server side backend structure and generate respective OpenAPI files. 
+In this approach full control over API lies within server side team - generated OpenAPI file is read only and cannot be effectively
+used to negotiate API between client and server.
+
+API first approach uses OpenAPI file as source of truth. Both client and server side generate code based on the OpenAPI file.
+
+## Recommended Packages
+
+List bellow provides comprehensive set of libraries that can be used for an end to end full stack application written in Node.JS, 
+Express.js as well as client side applications.
+
+### Code Generation Tools
+
+[openapitools/openapi-generator-cli](https://www.npmjs.com/package/@openapitools/openapi-generator-cli)
```

Comment:
We do not have standardized format for packages. Some docs use < some use (

Reply:
You mean in terms of the links?  If so we should probably choose one and try to be consistent. I always use []() but happy to go with what the overall consensus is.